### PR TITLE
Fix race conditions and clean up code in regression tests

### PR DESCRIPTION
**Pull request by spectranaut**
Test the CI

### CODE REVIEW

**Comment by spectranaut** [comment]
The two errors on this run of the CI are errors in the examples themselves, they have been reported:
https://github.com/w3c/aria-practices/issues/821
https://github.com/w3c/aria-practices/issues/785

**Comment by mcking65** [comment]
@tatermelon, @jessebeach, @sh0ji, are one of you available fore review?

**Comment by jessebeach** [comment]
@spectranaut the code changes just look like cleanup. Are there any behavior changes to review?

**Comment by spectranaut** [comment]
@jessebeach yup just clean up! No behavior changes.

File: test/util/assertTabOrder.js [hunk 1]
```diff
@@ -3,17 +3,13 @@
...
-  return t.context.session.executeScript(function () {
-    selector = arguments[0];
-    return document.activeElement === document.querySelector(selector);
-  }, selector);
+  return t.context.session.wait(async function () {
+    return t.context.session.executeScript(function () {
+      selector = arguments[0];
+      return document.activeElement === document.querySelector(selector);
+    }, selector);
+  }, 200);
```
Comment (sh0ji) [inline test/util/assertTabOrder.js]: Is there a reason this timeout is 200ms while the others are 500ms?

Comment (spectranaut) [reply test/util/assertTabOrder.js]: I put the 200ms waits before I stress tests but found 500 was better -- I'll switch them all to 500 for consistency, thanks for th catch!

**Comment by spectranaut** [comment]
hey @jessebeach the current failures are expected -- they surface errors in the tests themselves (see comment https://github.com/w3c/aria-practices/pull/837#issuecomment-413685577)

File: test/tests/tabs.js [hunk 2]
```diff
@@ -29,12 +29,20 @@ const openTabAtIndex = async function (t, tabOrderIndex) {
...
-const checkFocus = function (t, selector, index) {
-  return t.context.session.executeScript(function () {
-    const [selector, index] = arguments;
-    let items = document.querySelectorAll(selector);
-    return items[index] === document.activeElement;
-  }, selector, index);
+const waitAndCheckFocus = async function (t, selector, index) {
+  return t.context.session.wait(async function () {
+    return t.context.session.executeScript(function () {
+      const [selector, index] = arguments;
```
[the remark this answers is not in the document]
Comment (spectranaut) [reply test/tests/tabs.js]: hey @sh0ji this is actually a function that will be converted to a string then sent to the browser -- you can read about it here (ctrl-f executeScript): http://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_WebDriver.html

It's not executed in the context of this script and arguments can only be provided via the argument object.

**Comment by jessebeach** [comment]
> Hmmm now I have new failures I really don't understand -- something with the test set up: 

Maybe something to do with async/await functions?

**Comment by spectranaut** [comment]
@jessebeach -- this PR has some code and logic clean up, a couple race condition fixes (in my test code), and some lint error fixing. I'd like it to be accepted into the bocoup branch so I can work on top of these fixes. 

The errors in the CI are inconsistent and something I'd like to debug separately -- they are errors in the test infrastructure/coming from geckodriver. I'll have to enable logging and I'd like to do this on another PR.

**Pull request merged**